### PR TITLE
Update citation for Reactor module

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #23422

### CODE REVIEW

**Comment by GiudGiud** [comment]
@eshemon fyi

**Comment by eshemon** [comment]
thanks @GiudGiud!

File: modules/doc/content/citing.md [hunk 1]
```diff
@@ -291,19 +291,20 @@ journal = {Computational Mechanics},
...
-  year        = {2021},
-  month       = sep,
-  type        = {techreport},
-  doi         = {10.2172/1821454},
-  place       = {United States},
-  url         = {https://www.osti.gov/biblio/1821454},
+The following paper documents functionalities and demonstration of the MOOSE Reactor module.
+
+```
+@article{doi:10.1080/00295639.2022.2149231,
```
Comment (cticenhour) [inline modules/doc/content/citing.md:297]:
```suggestion
@article{shemon2023reactor,
```
To make the label more readable.

**Review by cticenhour — CHANGES_REQUESTED** [review]
One suggested change.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**